### ISSUE TITLE
ProofOnlySimpleIntegerVariableIDs break define if they are DirectOnly

### ISSUE THREAD

**Issue by mmcilree** [ciaranm/glasgow-constraint-solver#38]
If I define a `ProofOnlyIntegerVariableID` x as with`IntegerVariableProofRepresentation::DirectOnly` and then try to use that in a `define()` without first turning it into a literal I get a `map::at: key not found` exception. E.g.

```C++
auto x_po = propagators.create_proof_only_integer_variable(0_i, 10_i, "x", 
     IntegerVariableProofRepresentation::DirectOnly);

auto y_po = propagators.create_proof_only_integer_variable(0_i, 10_i, "y", 
     IntegerVariableProofRepresentation::DirectOnly);

propagators.define(initial_state, WeightedPseudoBooleanSum{} + 1_i * x_po - 1_i * y_po == 1);
```

I guess this is a "garbage in garbage out" kind of situation but perhaps we should throw a better exception?

**Comment by ciaranm** [comment]
Yeah, probably everywhere that says `_imp->integer_variable_bits.at` inside `proof.cc` could be changed to call a helper function.

**Comment by ciaranm** [comment]
Verified against current main with this issue's repro (two `DirectOnly` proof-only variables used as integer terms in a PB definition): the raw `map::at: key not found` is gone. Every bits lookup in `NamesAndIDsTracker` (`each_bit`, both `get_bit` overloads, `num_bits`, `negative_bit_coefficient`) now guards the map access and throws the typed `ProofError{"missing bits"}` at the point of misuse — the "helper everywhere that says `.at`" this issue asked for, in effect. `GCS_VERBOSE_LOGGING` gives a stacktrace when you need the where.

The message is terse — it names neither the variable nor the rule ("a DirectOnly variable has no bits encoding, so it cannot appear as an integer term in a pseudo-Boolean sum"). Accepting that as adequate for a garbage-in case and closing; threading the variable name into those five messages is a five-line change if it ever bites someone in practice.
